fms/models/mistral.py

```diff
@@ -0,0 +1,607 @@
+    config_params["rope_base"] = config.rope_theta
+    ig_params["src_vocab_size"] = config.vocab_size
+    config_params["nheads"] = config.num_attention_heads
+    config_params["nlayers"] = config.num_hidden_layers
+    config_params["hidden_grow_factor"] = inner_dim / config.hidden_size
+    config_params["tie_heads"] = config.tie_word_embeddings
+
+========= config.json -- https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3/blob/main/config.json
+  "attention_dropout": 0.0,
+  "bos_token_id": 1,          X
+  "eos_token_id": 2,          X
+  "hidden_act": "silu",
+  "hidden_size": 4096,
+  "initializer_range": 0.02,  X
+  "intermediate_size": 14336,
+  "max_position_embeddings": 32768,
+  "model_type": "mistral",
+  "num_attention_heads": 32,
+  "num_hidden_layers": 32,
+  "num_key_value_heads": 8,
+  "rms_norm_eps": 1e-05,
+  "rope_theta": 1000000.0,
+  "sliding_window": null,     X 
+  "tie_word_embeddings": false,
+  "torch_dtype": "bfloat16",
+  "transformers_version": "4.42.0.dev0",
+  "use_cache": true,
+  "vocab_size": 32768
+"""
+
+
+@dataclass
+class MistralConfig(ModelConfig):
+    src_vocab_size: int = 32768
+    nheads: int = 32
+    nlayers: int = 32
+    hidden_grow_factor: float = 14336 / 4096  # intermediate_size / hidden_size:emb_dim
+    multiple_of: int = 256  # borrowed from llama
+    tie_heads: bool = False
+    p_dropout: float = 0.0
+    activation_fn: str = "swish"
+    emb_dim: int = 4096
+    max_expected_seq_len: int = 32768
+    kvheads: int = 8
+    norm_eps: float = 1e-5
+    sliding_window: int = 4000
+    rope_base: float = 100_0000.0  # Same as rope_theta
+    fused_weights: bool = True  # FMS Specific -- For CPU/GPU = T, AIU = F
+    pad_id: int = -1  # borrowed from granite, we do need it
+    linear_config: Optional[Mapping[str, Any]] = None  # To suppor quantization
+
+
+_7b_config = MistralConfig()
+
+
+class MistralBlock(nn.Module):
+    def __init__(self, config: MistralConfig, rotary_emb: RotaryEmbedding):
+        super(MistralBlock, self).__init__()
+        self.config = config
+        emb_kq = self.config.emb_dim // self.config.nheads
+        emb_v = self.config.emb_dim // self.config.nheads
+
+        self.ln = LayerNormParameterized(
+            self.config.emb_dim,
+            elementwise_scale=True,
+            elementwise_shift=False,
+            use_mean=False,
+            eps=self.config.norm_eps,
+            use_high_precision_pow=True,
+        )
+        self.ff_ln = LayerNormParameterized(
+            self.config.emb_dim,
+            elementwise_scale=True,
+            elementwise_shift=False,
+            use_mean=False,
+            eps=self.config.norm_eps,
+            use_high_precision_pow=True,
+        )
+
+        if self.config.kvheads == 0:
+            kvheads = self.config.nheads
+        else:
+            kvheads = self.config.kvheads
+            assert self.config.nheads % self.config.kvheads == 0
+
+        self.attn = MultiHeadAttention(
```

Comment:
can you add a TODO comment here to add support sliding window attention in the future as right now we're ignoring it?